draft-ietf-tls-tls13.md

```diff
@@ -1787,6 +1787,9 @@ ClientHello (without modification) except:
   any PSKs which are incompatible with the server's indicated
   cipher suite.
 
+- Optionally adding, removing, or changing the length of the "padding"
+  extension {{RFC7685}}.
+
```

Comment:
Why not say it MUST be removed?

Reply:
Are the middleboxes that choke on particular sized ClientHellos not going to also choke on ClientHello2s?

Anyway, the whole point of the padding extension is to let you control record sizes on the wire; there seems to be no reason to restrict the available flexibility in one specific case.

Reply:
Older TLS versions already don't have two ClientHellos, so if a middlebox is actually getting to interpret the second one as a ClientHello, we probably failed to interop much earlier. (The padding extension was originally for a confused endpoint.)

But yeah, despite being more spec text, it seemed less work all around for implementations to retain the flexibility. One could imagine maybe a datagram transport wanting something funny to apply to both messages. (QUIC does pad ClientHellos, but I don't know the details to know if cookied ones are padded or if the padding extension is suitable. It's a little weird since it's got a discontinuity after N-4 bytes.)